### ISSUE TITLE
XML Content Type, Arbitrary JavaScript Payload, a fix ever deployed?

### ISSUE THREAD

**Issue by Montana** [NationalSecurityAgency/emissary#193]
There was an issue in 2020 where the HTTP response had an XML content type, but it did not prevent an attacker from executing an arbitrary JavaScript payload. My question is, was this a fix ever deployed for this? If not I'd be glad to deploy one. My idea is to limit `parse depth` and/or favor a SAX or iterparse-like parser for potential large data.

Is the recursion creating an actual, measured performance problem? If so, can you quantify the extent of recursion you are dealing with. `Antlr` is usually quite good at handling recursion.

That said, every instance of `expr` on the rule `rhs` will create a recursion. Grouping and limiting instances:

```antlr
expr : expr ('AND' | 'OR') expr 
     | 'a' | 'b' | 'c' | 'd'
     ; 
```
It won't change the number of recursions necessary to satisfy the rule -- that depends on the data being processed.

What might be a greater gain is, if the performance problem actually stems from a lot of deep recursion rule failures, is to restructure so you'll sooner or limit the number of rules (or subrules) potentially applicable for any given sequence of data.

See below: 

```java
@GET
@Path("/Document.action/{uuid}")
@Produces(MediaType.APPLICATION_XML)
 public Response documentShow(
       @Context HttpServletRequest request, 
       @PathParam("uuid") String uuid
     ) {
         final List<IBaseDataObject> payload = wsp.take(uuid);

         return Response.status(400).entity("<error>uuid " + uuid + " not found</error>").build();
  }
```
I really appreciate it! Thanks.

Cheers,
Montana Mendy

**Comment by drivenflywheel** [comment]
Hi Montana,

Yes, this issue was addressed in Emissary 6.0.0 by removing the affected endpoint.  No plans to bring it back.

Cheers,
D

**Comment by Montana** [comment]
Ah, thanks @drivenflywheel! It was unclear. I appreciate the quick answer.
